src/sbd-inquisitor.c

```diff
@@ -1307,37 +1303,50 @@ int main(int argc, char **argv, char **envp)
         } else if (strcmp(argv[optind], "ping") == 0) {
             exit_status = ping_via_slots(argv[optind + 1], servants_leader);
 
+        } else
+#endif
+        if (strcmp(argv[optind], "query-watchdog") == 0) {
+            exit_status = watchdog_info();
+        } else if (strcmp(argv[optind], "test-watchdog") == 0) {
+            exit_status = watchdog_test();
         } else if (strcmp(argv[optind], "watch") == 0) {
+            /* sleep $(sbd $SBD_DEVICE_ARGS dump | grep -m 1 msgwait | awk '{print $4}') 2>/dev/null */
+
+                const char *delay_source = "watchdog-timeout * 2";
+
+#if SUPPORT_SHARED_DISK
                 if(disk_count > 0) {
                     /* If no devices are specified, its not an error to be unable to find one */
                     open_any_device(servants_leader);
                 }
 
-                if (delay_start) {
-                    if (delay <= 0) {
-                        delay = get_first_msgwait(servants_leader);
-                    }
+                if (delay_start && delay <= 0) {
```

Comment:
Guess this should be in the block with (disc_count > 0) as get_first_msgwait is doomed to fail otherwise.

Reply:
Yep make sense.